drivers: regulator: support for regulator mode APIs in regulator shell

danieldegrasse opened the pull request:
Add support for regulator mode related APIs to regulator shell, so that the user can select new operation modes for the regulator and configure target voltages.

Additionally, clean up usage of `atoi`, as `strtol` is preferred. Remove `device_is_ready` checks, as `device_get_binding` will already perform this check.

danieldegrasse committed:
drivers: regulator: support for regulator mode APIs in regulator shell

Add support for regulator mode related APIs to regulator shell, so that
the user can select new operation modes for the regulator and configure
target voltages

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

carlescufi commented on drivers/regulator/regulator_shell.c:
```diff
@@ -109,6 +109,98 @@ static int cmd_set_ilim(const struct shell *sh, size_t argc, char **argv)
+		return -ENODEV;
+	}
+
+	mode = atoi(argv[2]);
```
This is just nitpicking, but `strto*` functions are preferred over `atoi` IIRC. @stephanosio

danieldegrasse replied:
Nits make code better :-). I've added a commit to this PR to cleanup all usages of `atoi` in this shell module, and switched the new commands to use `strtol` as well

stephanosio replied:
Indeed, `strto*` should be preferred over `atoi` because they provide better error handling.

bbilas commented on drivers/regulator/regulator_shell.c:
```diff
@@ -109,6 +109,98 @@ static int cmd_set_ilim(const struct shell *sh, size_t argc, char **argv)
+	const struct device *reg_dev;
+
+	reg_dev = device_get_binding(argv[1]);
+	if (!device_is_ready(reg_dev)) {
```
I guess you can do a simple NULL check just because `device_get_binding` has a check to see if the object was successfully initialized on startup.

danieldegrasse replied:
Thanks for the suggestion, I've switched to a `NULL` check. Also cleaned up usages of `device_get_binding` in the remainder of this shell module

danieldegrasse committed:
drivers: regulator: cleanup device_is_ready and atoi usage

Cleanup regulator shell to use strtol() over atoi(), as strtol features
error detection. Remove device_is_ready() checks, and replace them with
NULL checks as device_get_binding() calls device_is_ready() interally

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

carlescufi reviewed (approved):
LGTM
